config/config.go

```diff
@@ -659,6 +653,23 @@ func ParseConfigString(
 	return config, nil
 }
 
+// extractIamRole - extract IAM role details from Terragrunt flags block
+func extractIamRole(configString string, terragruntOptions *options.TerragruntOptions, filename string) error {
```

Comment:
NIT: Rename to `setIAMRole`, to indicate that this function has side effects (modifying attribute on `terragruntOptions`.